crates/router/src/core/payments/operations/payment_capture.rs

```diff
@@ -225,3 +259,58 @@ impl<F: Send + Clone> ValidateRequest<F, api::PaymentsCaptureRequest> for Paymen
         ))
     }
 }
+
+impl PaymentCapture {
+    async fn create_capture_and_update_attempt(
+        state: &AppState,
+        authorized_payment_attempt: storage::PaymentAttempt,
+        storage_scheme: enums::MerchantStorageScheme,
+    ) -> CustomResult<(storage::Capture, storage::PaymentAttempt), errors::StorageError> {
+        state
+            .store
+            .insert_capture(
+                Self::make_capture(&authorized_payment_attempt),
+                authorized_payment_attempt,
+                storage_scheme,
+            )
+            .await
+    }
+    fn make_capture(authorized_attempt: &storage::PaymentAttempt) -> storage::CaptureNew {
+        let capture_sequence = authorized_attempt.multiple_capture_count.unwrap_or(0) + 1;
+        let now = common_utils::date_time::now();
+        storage::CaptureNew {
+            payment_id: authorized_attempt.payment_id.clone(),
+            merchant_id: authorized_attempt.merchant_id.clone(),
+            capture_id: format!("{}_{}", authorized_attempt.attempt_id, capture_sequence),
+            status: enums::CaptureStatus::Started,
+            amount: authorized_attempt
+                .amount_to_capture
+                .unwrap_or(authorized_attempt.amount),
```

Comment:
Can you explain when this unwrap scenario will come? 

And amount_to_capture is equal amount is not true in all cases.